lib/web_ui/dev/integration_tests_manager.dart

```diff
@@ -0,0 +1,345 @@
+// found in the LICENSE file.
+
+import 'dart:io' as io;
+import 'package:path/path.dart' as pathlib;
+
+import 'common.dart';
+import 'environment.dart';
+import 'utils.dart';
+
+class IntegrationTestsManager {
+  final String _browser;
+
+  /// Installation directory for browser's driver.
+  ///
+  /// Always re-install since driver can change frequently.
+  /// It usually changes with each the browser version changes.
+  /// A better solution would be installing the browser and the driver at the
+  /// same time.
+  // TODO(nurhan): change the web installers to install driver and the browser
+  // at the same time.
+  io.Directory _driverDir;
+
+  IntegrationTestsManager(this._browser) {
+    _driverDir = io.Directory(
+        pathlib.join(environment.webUiRootDir.path, 'drivers', _browser));
+  }
+
+  Future<bool> run() async {
+    if (_browser != 'chrome') {
+      print('WARNING: integration tests are only suppoted on chrome for now');
+      return false;
+    } else {
+      bool driverReady = await prepareDriver();
+      // TODO(nurhan): provide a flag for running as if on CI. Also use this
+      // flag from LUCI.
+      if (isCirrus != null && isCirrus) {
+        // TODO(nurhan): fetch Flutter, we will use flutter pub get in the next steps.
+      }
+      if (driverReady) {
+        return await _runTests();
+      } else {
+        return false;
+      }
+    }
+  }
+
+  Future<bool> _cloneWebInstalllers() async {
+    final int exitCode = await runProcess(
+      'git',
+      <String>[
+        'clone',
+        'https://github.com/flutter/web_installers.git',
+      ],
+      workingDirectory: _driverDir.path,
+    );
+
+    if (exitCode != 0) {
+      io.stderr.writeln('ERROR: '
+          'Failed to clone web installers. Exited with exit code $exitCode');
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  Future<bool> _runPubGet(String workingDirectory) async {
+    // TODO(nurhan): Run flutter pub get from the fetched flutter on CI.
+    final int exitCode = await runProcess(
+      //environment.pubExecutable,
+      'flutter',
+      <String>[
+        'pub',
+        'get',
+      ],
+      workingDirectory: workingDirectory,
+    );
+
+    if (exitCode != 0) {
+      io.stderr.writeln(
+          'ERROR: Failed to run pub get. Exited with exit code $exitCode');
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  Future<bool> _runDriver() async {
+    final int exitCode = await runProcess(
+      environment.dartExecutable,
+      <String>[
+        'lib/web_driver_installer.dart',
+        '${_browser}driver',
+        '--install-only',
+      ],
+      workingDirectory: pathlib.join(
+          _driverDir.path, 'web_installers', 'packages', 'web_drivers'),
+    );
+
+    if (exitCode != 0) {
+      io.stderr.writeln(
+          'ERROR: Failed to run driver. Exited with exit code $exitCode');
+      return false;
```

Comment:
I'd prefer that we throw exceptions and halt the program as soon as we encounter irrecoverable errors like this (the project structure validation falls in the same category as well). This way you'll have a stack trace to pinpoint the code right away. Otherwise, this code will return `false` and keep on trucking, potentially causing more errors down the line.

Reply:
Same comment:

If I throw, and if the user is using
felt test
which runs both unit and integration tests, the process will terminate. I think this makes sense only if the developer is using:
felt test --integration-tests-only

I personally believe it is better to have as much as results possible when you run tests before the process fails.

Reply:
done. I changed all the booleans to exceptions. resolving the comment.